randovania/gui/lib/signal.py

```diff
@@ -0,0 +1,17 @@
+import typing
+from collections.abc import Callable
+from typing import Generic, TypeVarTuple
+
+FuncArgs = TypeVarTuple("FuncArgs")
+
+
+class RdvSignal(Generic[*FuncArgs]):
+    def __init__(self) -> None:
+        self.subscribers: list = []
+
+    def emit(self, *func_args: *FuncArgs) -> None:
+        for x in self.subscribers:
+            x(*func_args)
+
+    def connect(self, func: Callable[[*FuncArgs], typing.Any]) -> None:
```

Comment:
return value should be None

Probably a good idea to make a type alias for the callable

Reply:
Do you know how to do it?
If I just add something like `Function = Callable[[*FuncArgs], None]` next to the `FuncArgs` it doesn't really work for mypy. I guess because it doesn't know the params for the generic if you type alias outside of the class.
So, you get something like `Argument 1 to "connect" of "RdvSignal" has incompatible type "Callable[[str], None]"; expected "Callable[[VarArg(Any)], None]" `. 

Define it within the class doesn't work for me `Generic type alias within class cannot use bound type variables FuncArgs`

Reply:
Hm.. yeah that can be a problem. It could potentially be a Protocol, but I don't think it helps much.
I imagine explicitly marking it as `TypeAlias` doesn't help?